packages/shared/src/hooks/usePostById.ts

```diff
@@ -91,6 +92,17 @@ export const removePostComments = (
   });
 };
 
+export const postByIdQueryOptions = ({ id }: { id: string }) => {
+  return {
+    queryKey: getPostByIdKey(id),
+    queryFn: async () => {
+      const res = await gqlClient.request<PostData>(POST_BY_ID_QUERY, { id });
+
+      return res;
+    },
+  };
+};
```

Comment:
Instead of building custom hooks each time to fetch specific data this way we export all `useQuery` options in this creator function which we can then reuse in new hooks to get types automatically in queries, mutations and updates.